### PR TITLE
in-addr-util: fix in_addr_prefix_nth() and use the function to get ip address range in firewall-util

### PR DESCRIPTION
**Pull request by yuwata**
Follow-up for #18007.

### CODE REVIEW

File: src/shared/firewall-util-nft.c [hunk 1]
```diff
@@ -848,63 +844,27 @@ static int fw_nftables_recreate_table(sd_netlink *nfnl, int af, sd_netlink_messa
...
-                uint16_t overflow = ret_end->s6_addr[j] + carry;
-
-                ret_end->s6_addr[j] = overflow;
-                if (overflow <= 0xff)
-                        break;
-                carry = 1;
-        }
-
-        if (memcmp(ret_start, ret_end, sizeof(*ret_start)) > 0)
-                zero(ret_end);
```
Comment (yuwata) [inline src/shared/firewall-util-nft.c:888]: @fw-strlen Is this important? In this PR, `in_addr_prefix_range()` refuses such situation.

Comment (fw-strlen) [reply src/shared/firewall-util-nft.c:888]: Right, its safe to remove this.  It was inherited from the ipv4 version (where its needed). Changes LGTM, thanks!

Comment (yuwata) [reply src/shared/firewall-util-nft.c:888]: Thank you for the confirmation.

**Comment by yuwata** [comment]
Setting the green label as @fw-strlen accepts this.

**Pull request merged**